include/seastar/core/estimated_histogram.hh

```diff
@@ -0,0 +1,385 @@
+ * infinity bucket (the assumption is that it can happen but it is rare).
+ *
+ * Note the difference between the first and last buckets.
+ * The first bucket is just like a regular bucket but has a second roll to collect unexpected low values.
+ * The last bucket, also known as the infinity bucket, collect all values that passes the defined Max,
+ * it only collect those values.
+ *
+ * Buckets Distribution (limits)
+ * =============================
+ * The buckets limits in the histogram are defined similar to a floating-point representation.
+ *
+ * Buckets limits have an exponent part and a linear part.
+ *
+ * The exponential part is a power of 2. Each power-of-2 range [2^n..2^n+1)
+ * is split linearly to 'Precision' number of buckets.
+ *
+ * The total number of buckets is:
+ *   NUM_BUCKETS = log2(Max/Min)*Precision +1
+ *
+ * For example, if the Min value is 128, the Max is 1024 and the Precision is 4, the number of buckets is 13.
+ *
+ * Anything below 160 will be in the bucket 0, anything above 1024 will be in bucket 13.
+ * Note that the first bucket will include all values below Min.
+ *
+ * the range [128, 1024) will be split into log2(1024/128) = 3 ranges:
+ * 128, 256, 512, 1024
+ * Or more mathematically: [128, 256), [256, 512), [512,1024)
+ *
+ * Each range is split into 4 (The Precision).
+ * 128            | 256            | 512            | 1024
+ * 128 160 192 224| 256 320 384 448| 512 640 768 896|
+ *
+ * To get the exponential part of an index you divide by the Precision.
+ * The linear part of the index is Modulus the precision.
+ *
+ * Calculating the bucket lower limit of bucket i:
+ * The exponential part: exp_part = 2^floor(i/Precision)* Min
+ *    with the above example 2^floor(i/4)*128
+ * The linear part: (i%Precision) * (exp_part/Precision)
+ *    With the example: (i%4) * (exp_part/4)
+ *
+ * So the lower limit of bucket 6:
+ *    2^floor(6/4)* 128  = 256
+ *    (6%4) * 256/4 = 128
+ *    lower-limit   = 384
+ *
+ * How to find a bucket index for a value
+ * =======================================
+ * The bucket index consist of two parts:
+ * higher bits (exponential part) are based on log2(value/min)
+ *
+ * lower bits (linear part) are based on the 'n' MSB (ignoring the leading 1) where n=log2(Precision).
+ * Continuing with the example where the number of precision bits: PRECISION_BITS = log2(4) = 2
+ *
+ * for example: 330 (101001010)
+ * The number of precision_bits: PRECISION_BITS = log2(4) = 2
+ * higher bits: log2(330/128) = 1
+ * MSB: 01 (the highest two bits following the leading 1)
+ * So the index: 101 = 5
+ *
+ * About the Min, Max and Precision
+ * ================================
+ * For Min, Max and Precision, choose numbers that are a power of 2.
+ *
+ * Limitation: You must set the MIN value to be higher or equal to the Precision.
+ *
+ */
+template<uint64_t Min, uint64_t Max, size_t Precision>
+class approx_exponential_histogram {
```

Comment:
s/approx/approximate/

Reply:
Seastar naming convention is to use full names. It shouldn't be affected by the donor repository.

If ScyllaDB wants to use it, it can use a type synonym, or update the mentions.